Client/Frontend/Widgets/DrawerViewController.swift

```diff
@@ -0,0 +1,265 @@
+            let cornerRadii = CGSize(width: DrawerViewControllerUX.DrawerCornerRadius, height: DrawerViewControllerUX.DrawerCornerRadius)
+            let path = UIBezierPath(roundedRect: bounds, byRoundingCorners: [.topLeft, .topRight], cornerRadii: cornerRadii)
+            shapeLayer.frame = bounds
+            shapeLayer.path = path.cgPath
+            layer.mask = shapeLayer
+        }
+    }
+}
+
+public class DrawerViewController: UIViewController {
+    public let childViewController: UIViewController
+
+    public var drawerView: DrawerView!
+    public var drawerViewAnimatedConstraint: Constraint?
+
+    fileprivate var backgroundOverlayView: UIView!
+    fileprivate var handleView: UIView!
+
+    // Only used in iPad layout.
+    fileprivate var xPosition: CGFloat = 0 {
+        didSet {
+            let maxX = view.frame.minX
+
+            if xPosition < maxX - DrawerViewControllerUX.DrawerPadWidth {
+                xPosition = maxX - DrawerViewControllerUX.DrawerPadWidth
+            } else if xPosition > maxX {
+                xPosition = maxX
+            }
+
+            backgroundOverlayView.alpha = (xPosition + DrawerViewControllerUX.DrawerPadWidth) / DrawerViewControllerUX.DrawerPadWidth / 2
+            drawerViewAnimatedConstraint?.update(offset: xPosition)
+            view.layoutIfNeeded()
+        }
+    }
+
+    // Only used in non-iPad layout.
+    fileprivate var yPosition: CGFloat = DrawerViewControllerUX.DrawerTopStop {
+        didSet {
+            let maxY = view.frame.maxY
+
+            if yPosition < DrawerViewControllerUX.DrawerTopStop {
+                yPosition = DrawerViewControllerUX.DrawerTopStop
+            } else if yPosition > maxY {
+                yPosition = maxY
+            }
+
+            backgroundOverlayView.alpha = (maxY - yPosition) / maxY / 2
+            drawerViewAnimatedConstraint?.update(offset: yPosition)
+            view.layoutIfNeeded()
+        }
+    }
+
+    fileprivate var showingPadLayout: Bool {
+        return traitCollection.userInterfaceIdiom == .pad &&
+            traitCollection.horizontalSizeClass == .regular
+    }
+
+    public init(childViewController: UIViewController) {
+        self.childViewController = childViewController
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+
+        let backgroundOverlayView = UIView()
+        self.backgroundOverlayView = backgroundOverlayView
+        backgroundOverlayView.backgroundColor = .black
+        backgroundOverlayView.alpha = 0
+        view.addSubview(backgroundOverlayView)
+
+        let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(didRecognizeTapGesture))
+        backgroundOverlayView.addGestureRecognizer(tapGestureRecognizer)
+
+        let drawerView = DrawerView()
```

Comment:
if the ` DrawerView()` is created in viewdidload we could create it at the top there in order to avoid the ` DrawerView!`